features/machine/machine.feature

```diff
@@ -386,19 +386,21 @@ Feature: Machine features testing
     Then admin ensures "<name>" machine_set_machine_openshift_io is deleted after scenario
 
     Given I obtain test data file "cloud/ms-aws/<file_name>"
     When I run oc create over "<file_name>" replacing paths:
-      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]           | <%= infrastructure("cluster").infra_name %> |
-      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["iamInstanceProfile"]["id"]         | <%= cb.default_iamInstanceProfile %>        |
-      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"]        | <name>                                      |
-      | ["spec"]["template"]["metadata"]["labels"]["machine.openshift.io/cluster-api-cluster"]    | <%= infrastructure("cluster").infra_name %> |
-      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["ami"]["id"]                        | <%= cb.default_ami_id %>                    |
-      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["placement"]["availabilityZone"]    | <%= cb.default_availability_zone %>         |
-      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["subnet"]["filters"][0]["values"]   |  <%= cb.default_subnet[0].values[1] %>      |
-      | ["spec"]["template"]["metadata"]["labels"]["machine.openshift.io/cluster-api-machineset"] | <name>                                      |
-      | ["metadata"]["name"]                                                                      | <name>                                      |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]                    | <%= infrastructure("cluster").infra_name %>                    |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["iamInstanceProfile"]["id"]                  | <%= cb.default_iamInstanceProfile %>                           |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"]                 | <name>                                                         |
+      | ["spec"]["template"]["metadata"]["labels"]["machine.openshift.io/cluster-api-cluster"]             | <%= infrastructure("cluster").infra_name %>                    |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["ami"]["id"]                                 | <%= cb.default_ami_id %>                                       |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["securityGroups"][0]["filters"][0]["values"] | <%= cb.default_secgrp[0].values[0].to_s.split(",")[1][11..-1]%>|
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["placement"]["availabilityZone"]             | <%= cb.default_availability_zone %>                            |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["subnet"]["filters"][0]["values"]            | <%= cb.default_subnet[0].values[1] %>                          |
```

Comment:
For the cluster if the machineset's subnet is in form of ID, will it fail?

Reply:
Got it, thanks @miyadav we can merge this first to see if it still fails in some clusters. If it still fails, we can use golang. It is not easy to check whether the subnet is id or filter in cucushift.
It seems that when the vpc is not created by the installer, but create it ourselves, and then write the vpcid to install-config, the subnet will be in the form of id.  The template name contains customer-vpc, seems the subnet is id.